device_arm64.go

```diff
@@ -6,13 +6,37 @@
 
 package main
 
-const (
-	rootBusPath = "/devices/platform/4010000000.pcie/pci0000:00"
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+)
 
+const (
 	// From https://www.kernel.org/doc/Documentation/acpi/namespace.txt
 	// The Linux kernel's core ACPI subsystem creates struct acpi_device
 	// objects for ACPI namespace objects representing devices, power resources
 	// processors, thermal zones. Those objects are exported to user space via
 	// sysfs as directories in the subtree under /sys/devices/LNXSYSTM:00
 	acpiDevPath = "/devices/LNXSYSTM"
 )
+
+func createRootBusPath() string {
+	startRootBusPath := "/devices/platform"
+	endRootBusPath := "/pci0000:00"
+	sysStartRootBusPath := filepath.Join(sysfsDir, startRootBusPath)
+	files, err := ioutil.ReadDir(sysStartRootBusPath)
+	if err != nil {
+		fmt.Errorf("Error reading %s: %s", sysStartRootBusPath, err)
+	}
+
+	// find out the director end with ".pcie"
+	for _, file := range files {
+		if strings.HasSuffix(file.Name(), ".pcie") {
```

Comment:
If the comment above has a typo, shouldn't this also test that `file.IsDir() == true`?